mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ServerState.scala

```diff
@@ -596,6 +598,21 @@ import scala.util.{Failure, Success}
                 data.settings
               )
             )
+          case (_, ClientConnection.ConnectionLost) =>
+            clientDisconnected(
+              Disconnected(
+                data.publishers,
+                data.activeConsumers,
+                data.activeProducers,
+                data.pendingLocalPublications,
+                data.pendingRemotePublications,
+                data.consumerPacketRouter,
+                data.producerPacketRouter,
+                data.publisherPacketRouter,
+                data.unpublisherPacketRouter,
+                data.settings
+              )
+            )
```

Comment:
Same for the server - we now move to the disconnected state if we have lost the connection.